diffrax/_brownian/path.py

```diff
@@ -124,19 +124,19 @@ def _evaluate_leaf(
         t1: RealScalarLike,
         key,
         shape: jax.ShapeDtypeStruct,
-        levy_area: type[Union[BrownianIncrement, SpaceTimeLevyArea]],
+        levy_area: type[AbstractBrownianIncrement],
         use_levy: bool,
     ):
         w_std = jnp.sqrt(t1 - t0).astype(shape.dtype)
         w = jr.normal(key, shape.shape, shape.dtype) * w_std
-        dt = t1 - t0
+        dt = jnp.asarray(t1 - t0, dtype=shape.dtype)
 
-        if levy_area is SpaceTimeLevyArea:
+        if issubclass(levy_area, AbstractSpaceTimeLevyArea):
```

Comment:
Fair, I'll try to make these changes. Although, I never directly instantiate `self.levy_area`. Instead I did
```python
if issubclass(levy_area, AbstractSpaceTimeLevyArea):
  ...
  return SpaceTimeLevyArea(...)
```
which is perfectly fine. But fair, I will change it nonetheless.

This did make it a bit harder to do things like the following (in `helpers.py`):
```python
levy_area1 = get_minimal_la(solver)
levy_area2 = get_minimal_la(ref_solver)
# Stricter levy_area requirements inherit from less strict ones
levy_area = levy_area1 if issubclass(levy_area1, levy_area2) else levy_area2
```
I could not do this with non-abstract types since inheritance does not provide an ordering on the set {`BrownianIncrement`, `SpaceTimeLevyArea`, `SpaceTimeTimeLevyArea`}.

I got around it by defining
```python
def abstract_la_to_la(abstract_la):
    if issubclass(abstract_la, diffrax.AbstractSpaceTimeTimeLevyArea):
        return diffrax.SpaceTimeTimeLevyArea
    elif issubclass(abstract_la, diffrax.AbstractSpaceTimeLevyArea):
        return diffrax.SpaceTimeLevyArea
    elif issubclass(abstract_la, diffrax.AbstractBrownianIncrement):
        return diffrax.BrownianIncrement
    else:
        raise ValueError(f"Unknown levy area {abstract_la}")
```

Reply:
> I could not do this with non-abstract types since inheritance does not provide an ordering on the set {BrownianIncrement, SpaceTimeLevyArea, SpaceTimeTimeLevyArea}.

Ah! This is again a really interesting point of discussion.

The scenario you describe here is very similar to the dtype promotion rules in NumPy. For example, `int32` and `float32` do not have any ordering specified via subclass relationships, but there is nonetheless a sense in which "`float32` is bigger than `int32`".

The usual way to fix this is to do the same as is done for dtypes: introduce a lattice, for which the [join](https://en.wikipedia.org/wiki/Join_and_meet) is the "largest" type that should be used. (Which may not be amongst those considered, e.g. `join(int64, float32) == float64`.)

Now FWIW introducing a lattice over Brownian types here is *definitely* overkill so definitely please do something else instead! I just thought you might find it interesting to know about.

Reply:
That makes sense, thanks!

And yes, I settled for something simpler.